tests/integration/test_xrdsst.py

```diff
@@ -0,0 +1,127 @@
+import unittest
+import time
+import subprocess
+import os
+import json
+import git
+import docker
+import urllib3
+
+from definitions import ROOT_DIR
+from xrdsst.controllers.base import BaseController
+from xrdsst.controllers.init import InitServerController
+from xrdsst.controllers.timestamp import TimestampController
+from xrdsst.controllers.token import TokenController
+from xrdsst.main import XRDSSTTest
+
+
+class TestXRDSST(unittest.TestCase):
+
+    git_repo = 'https://github.com/nordic-institute/X-Road.git'
+    local_folder = os.path.join(ROOT_DIR, "tests/integration/X-Road")
+    branch_name = 'develop'
+    docker_folder = local_folder + '/Docker/securityserver'
+    image = 'xroad-security-server:latest'
+    url = 'https://localhost:4000/api/v1/api-keys'
+    roles = '[\"XROAD_SYSTEM_ADMINISTRATOR\",\"XROAD_SECURITY_OFFICER\"]'
+    header = 'Content-Type: application/json'
+    max_retries = 10
+    curl_retry_wait_seconds = 5
+    api_key = None
+
+    @staticmethod
+    def single_ss_config():
+        configuration_anchor = os.path.join(ROOT_DIR, "tests/resources/configuration-anchor.xml")
+        config = {
+            'logging': [{'file': '/var/log/xrdsst_test.log', 'level': 'INFO'}],
+            'security-server':
+                [{'name': 'ss',
+                  'url': 'https://localhost:8000/api/v1',
+                  'api_key': 'X-Road-apikey token=a2e9dea1-de53-4ebc-a750-6be6461d91f0',
+                  'configuration_anchor': configuration_anchor,
+                  'owner_dn_country': 'EE',
+                  'owner_dn_org': 'ORG',
+                  'owner_member_class': 'GOV',
+                  'owner_member_code': '1234',
+                  'security_server_code': 'SS',
+                  'software_token_id': 0,
+                  'software_token_pin': '1234'}]}
+        return config
+
+    def setUp(self):
+        config = self.single_ss_config()
+        name = config["security-server"][0]["name"]
+        if not os.path.exists(self.local_folder):
+            git.Repo.clone_from(self.git_repo, self.local_folder, branch=self.branch_name)
```

Comment:
Always cloning ``X-Road`` repository should be avoided, repositories tend to only grow larger with time.

Reply:
I do not see any other way. One option would be just to download the Dockerfile manually and include to the project resources, but that can create issues when the file is updated in the repo, hence we will not get the updates. Cloning the repo will always provide us the latest Dockerfile. In other places the repo is also cloned, check deployment scripts of X-Road, it is cloned also when testing in Jenkins.

Reply:
I will present some draft ideas for considering possible approaches to this:

1. Using X-Road project as submodule in the main folder, cloned at first, updated later, making use of its relevant test resources during integration test cycle.

1. Using GitHub branch/tag URL reference for obtaining up-to-date copy of ``Dockerfile``, e.g. for ``develop`` it could be https://raw.githubusercontent.com/nordic-institute/X-Road/develop/Docker/securityserver/Dockerfile